### PR TITLE
Support re-requesting when we are outside of the randomness availability window

### LINKED ISSUE #1815: Users should be able to re-request attestations if they are out of the randomness window
### Expected Behavior

- You should be able to re-request

### Current Behavior

You can't re-request until the attestation has expired

### PR DESCRIPTION
**Pull request by nambrot**
### Description

We currently forced users to wait for an attestation to expire if they could not select for an unselected request due to them being outside the randomness availability window. This PR allows for a user to overwrite the unselected request since it is effectively useless at that point.

I also increased the randomness availability window to match the attestation expiry in any case

### Tested

- Unit tests

### Related issues

- Fixes #1815 

### Backwards compatibility

- Yes

### CODE REVIEW

**Review by asaj — COMMENTED** [review]
Why don't we just couple these, so that we keep randomness around until the attestation request is expired?

File: packages/protocol/contracts/identity/Attestations.sol [hunk 1]
```diff
@@ -601,4 +602,12 @@ contract Attestations is
     return (attestation.status == AttestationStatus.Incomplete &&
       !isAttestationExpired(attestation.blockNumber));
   }
+
+  function isAttestationRequestSelectable(uint256 attestationRequestBlock)
+    internal
+    view
+    returns (bool)
+  {
+    return block.number < attestationRequestBlock.add(getRandom().randomnessBlockRetentionWindow());
```
Comment (timmoreton) [inline packages/protocol/contracts/identity/Attestations.sol:611]: strictly also `block.number >= attestationRequestBlock + 4 && ` ? right?

File: packages/protocol/contracts/identity/Attestations.sol [hunk 2]
```diff
@@ -166,7 +166,8 @@ contract Attestations is
 
     require(
       state.unselectedRequests[msg.sender].blockNumber == 0 ||
-        isAttestationExpired(state.unselectedRequests[msg.sender].blockNumber),
+        isAttestationExpired(state.unselectedRequests[msg.sender].blockNumber) ||
+        !isAttestationRequestSelectable(state.unselectedRequests[msg.sender].blockNumber),
       "There exists an unexpired, unselected attestation request"
```
Comment (timmoreton) [inline packages/protocol/contracts/identity/Attestations.sol:171]: update msg: `that can be selected` ?

**Review by timmoreton — APPROVED** [review]

File: packages/protocol/contracts/identity/Attestations.sol (hunk 1 above)
Comment (nambrot) [reply packages/protocol/contracts/identity/Attestations.sol:611]: Strictly yes, but `4` should almost always be smaller

**Pull request merged**